walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,144 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
 
 public final class FindMeetingQuery {
+  public static final int TOTAL_SLOTS = 24 * 60;
+  public static final int DUMMY_DURATION = 1;
+
+  public Collection<TimeRange> query1(Collection<Event> events, MeetingRequest request) {
+    long duration = request.getDuration();
+    Collection<TimeRange> answer = new ArrayList<>();
+
+    if (duration > TOTAL_SLOTS) {
+      return answer;
+    }
+
+    // build in O(PlogP) so contains() run in O(logP) instead of O(P), where P is the max number of Attendees per Event
+    TreeSet<String> attendees = new TreeSet(request.getAttendees());
```

Comment:
How about use hash-map so that contains is O(1)?

Reply:
right, will change and update the time complexity, missed that attendees do not require sorting